app/src/app/store/middlewareConfig.ts

```diff
@@ -0,0 +1,24 @@
+import {DevtoolsOptions, PersistOptions} from 'zustand/middleware';
+import {MapStore} from './mapStore';
+
+export const persistOptions: PersistOptions<MapStore, Partial<MapStore>> = {
+  name: 'districtr-persistrictr',
+  version: 0,
+  partialize: state => ({
+    userMaps: state.userMaps,
+  }),
+};
+
+export const devToolsConfig: DevtoolsOptions = {
+  serialize: {
+    options: {
+      set: (setInStore: Set<unknown>) => Array.from(setInStore),
+      map: (mapInStore: Map<string, unknown>) => {
```

Comment:
I believe these will run on every change to the state, yes. Or, at least the state change involving that part of the state?

For user-facing perf concerns, we set up the devtools wrapper to only be in local dev, no production deploy.

Reply:
> Naïve question @nofurtherinformation but what is the benefit of serializing the state in our case?

Apologies missed this question! Without serialization configuration for dev tools, we are unable to see the state in redux dev tools, which makes observing state in complex scenarios way, way easier

Without serialization config:
![Screenshot 2024-11-18 at 4 35 34 PM](https://github.com/user-attachments/assets/8900e80b-c90f-4897-a9ab-411d2b710345)

With serialization config:
![Screenshot 2024-11-18 at 4 36 56 PM](https://github.com/user-attachments/assets/2e87e414-bdfa-4207-bbb7-7312245dcda3)

This may seem like minor devex, but will make debugging substantially better.